src/aux-records/XpController.ts

```diff
@@ -0,0 +1,176 @@
+ */
+export class XpController {
+    private _auth: AuthController;
+    private _authStore: AuthStore;
+    private _xpStore: XpStore;
+
+    constructor(config: XpConfig) {
+        this._auth = config.authController;
+        this._authStore = config.authStore;
+        this._xpStore = config.xpStore;
+    }
+
+    /**
+     * Generate an account (user or contract) configured for the given currency
+     * @param currency The currency to configure the account for (default: USD)
+     */
+    private _generateAccount(currency: keyof ISO4217_Map = 'USD'): XpAccount {
+        return {
+            id: uuid(),
+            currency,
+            createdAtMs: Date.now(),
+            updatedAtMs: Date.now(),
+            closedTimeMs: null,
+        };
+    }
+
+    /**
+     * Create an XP user for the given auth user
+     * @param authUserId The ID of the auth user to create an XP user for
+     */
+    @traced(TRACE_NAME)
+    private async _createXpUser(
+        authUserId: string
+    ): Promise<CreateXpUserResult> {
+        return await tryScope(
+            async () => {
+                const authUser = await this._authStore.findUser(authUserId);
+                if (!authUser) {
+                    return {
+                        success: false,
+                        errorCode: 'user_not_found',
+                        errorMessage: 'The user was not found.',
+                    };
+                }
+
+                const xpId = uuid();
+                const account: XpAccount = this._generateAccount();
+                const user: XpUser = {
+                    id: xpId,
+                    userId: authUserId,
+                    accountId: account.id,
+                    requestedRate: null,
+                    createdAtMs: Date.now(),
+                    updatedAtMs: Date.now(),
+                };
+                await this._xpStore.createXpUserWithAccount(user, account);
+                return { success: true, user };
+            },
+            {
+                scope: [TRACE_NAME, this._createXpUser.name],
+                errMsg: 'An error occurred while creating the user or associated account.',
+                returnOnError: {
+                    success: false,
+                    errorCode: 'server_error',
+                    errorMessage:
+                        'An error occurred while creating the user or associated account.',
+                },
+            }
+        );
+    }
+
+    /**
+     * Get an Xp user's meta data (Xp meta associated with an auth user)
+     * Creates an Xp user for the auth user if one does not exist
+     */
+    async getXpUser(id: GetXpUserById): Promise<GetXpUserResult> {
+        return await tryScope(
+            async () => {
+                let user =
+                    (await this._xpStore[
+                        `getXpUserBy${id.userId ? 'Auth' : ''}Id`
+                    ](id.userId ?? id.xpId)) ?? null;
```

Comment:
@LandonSiler Please prefer regular dot-syntax for choosing which store function to call. IMO it will be easier to read and also won't require constructing a new string just to call a function:

```typescript
let user = id.userId ?
    await this._xpStore.getXpUserByAuthId(id.userId) : 
    await this._xpStore.getXpUserById(id.xpId);
```

Reply:
Absolutely 👍, I believe my initial plan with the dynamic accessing was to support multiple forms of Id querying which weren't implemented in any capacity outside of auth and xp Ids', this is/was a YAGNI moment :)